steamclog/src/main/java/com/steamclock/steamclog/Destinations.kt

```diff
@@ -201,23 +215,34 @@ internal class ExternalLogFileDestination : Timber.DebugTree() {
     private fun isCachedLogFileValid(expiryMs: Long): Boolean {
         // if you have a cached log file and you checked it within the expiry window
         val now = Date().time
-        val currentLogFileCachedTime = currentLogFileCachedTime
-        return cachedCurrentLogFile != null &&
+        val currentLogFileCachedTime = currentLogFileEstimatedCreatedTime
+        return currentLogFileEstimatedCreatedTime != null &&
                 currentLogFileCachedTime != null &&
                 now - currentLogFileCachedTime < expiryMs
     }
 
     private fun updateCachedLogFile(file: File?) {
-        cachedCurrentLogFile = file
-        currentLogFileCachedTime = Date().time
+        // Only update if we are changing files; this helps us simulate when a file was "created" so
+        // that we can cycle every 10 mins
+        if (currentLogFile == file) { return }
```

Comment:
Not checking for this was causing the `currentLogFileEstimatedCreatedTime` value to be incorrectly updated each time, meaning that the expiry calculation couldn't succeed as expected if the app was continually writing logs.